File: workshops/agentcore-workday-integration/dashboard/ac_client.py
import boto3
import os
import logging
import json
import uuid
from user_auth import USER_ID

l = logging.getLogger("aws.agentcore_client")

ac_client = boto3.client("bedrock-agentcore")

AGENT_RUNTIME_ARN = os.getenv("AGENT_RUNTIME_ARN")
SESSION_ID=str(uuid.uuid4())
l.info(f"AGENT_RUNTIME_ARN={AGENT_RUNTIME_ARN}")
l.info(f"SESSION_ID={SESSION_ID}")

def invoke_agent(payload):
    l.info("> invoke_agent")
    payload["user_id"] = USER_ID
    
    response = ac_client.invoke_agent_runtime(
        agentRuntimeArn = AGENT_RUNTIME_ARN,
        payload=json.dumps(payload),
        contentType="application/json",
        runtimeSessionId=SESSION_ID,
        runtimeUserId=USER_ID
    )
    status_code = response["ResponseMetadata"]["HTTPStatusCode"]
    l.info(f"| response.status_code={status_code}")
    response_body = response["response"].read().decode()
    # print(f">| response_body: {response_body}")
    return json.loads(response_body)

File: workshops/agentcore-workday-integration/dashboard-js/ac_client.js
import { randomUUID } from "crypto";
import { BedrockAgentCoreClient, InvokeAgentRuntimeCommand } from "@aws-sdk/client-bedrock-agentcore";
import { USER_ID } from "./user_auth.js";

const AGENT_RUNTIME_ARN = process.env.AGENT_RUNTIME_ARN;

if (!AGENT_RUNTIME_ARN) {
  console.error("ERROR: AGENT_RUNTIME_ARN environment variable is not set");
  process.exit(1);
}

const client = new BedrockAgentCoreClient();
export const SESSION_ID = randomUUID();

console.log(`AGENT_RUNTIME_ARN=${AGENT_RUNTIME_ARN}`);
console.log(`SESSION_ID=${SESSION_ID}`);

export async function invokeAgent(payload) {
  payload.user_id = USER_ID;
  const cmd = new InvokeAgentRuntimeCommand({
    agentRuntimeArn: AGENT_RUNTIME_ARN,
    payload: JSON.stringify(payload),
    contentType: "application/json",
    runtimeSessionId: SESSION_ID,
    runtimeUserId: USER_ID,
  });
  const response = await client.send(cmd);
  const chunks = [];
  for await (const chunk of response.response) chunks.push(chunk);
  return JSON.parse(Buffer.concat(chunks).toString());
}
